offchain/rollups-events/src/broker/mod.rs

```diff
@@ -28,10 +32,55 @@ pub mod indexer;
 
 pub const INITIAL_ID: &'static str = "0";
 
+#[derive(Clone)]
+enum BrokerConnection {
+    ConnectionManager(ConnectionManager),
+    ClusterConnection(ClusterConnection),
+}
+
+impl ConnectionLike for BrokerConnection {
+    fn req_packed_command<'a>(
+        &'a mut self,
+        cmd: &'a Cmd,
+    ) -> RedisFuture<'a, Value> {
+        match self {
+            Self::ConnectionManager(connection) => {
+                connection.req_packed_command(cmd)
+            }
+            Self::ClusterConnection(connection) => {
+                connection.req_packed_command(cmd)
+            }
+        }
+    }
+
+    fn req_packed_commands<'a>(
+        &'a mut self,
+        cmd: &'a Pipeline,
+        offset: usize,
+        count: usize,
+    ) -> RedisFuture<'a, Vec<Value>> {
+        match self {
+            Self::ConnectionManager(connection) => {
+                connection.req_packed_commands(cmd, offset, count)
+            }
+            Self::ClusterConnection(connection) => {
+                connection.req_packed_commands(cmd, offset, count)
+            }
+        }
+    }
+
+    fn get_db(&self) -> i64 {
+        match self {
+            Self::ConnectionManager(connection) => connection.get_db(),
+            Self::ClusterConnection(connection) => connection.get_db(),
+        }
+    }
+}
+
 /// Client that connects to the broker
 #[derive(Clone)]
 pub struct Broker {
-    connection: ConnectionManager,
+    connection: BrokerConnection,
```

Comment:
Couldn't this be `Box<dyn ConnectionLike>` instead? If yes, you wouldn't need the BrokerConnection struct.

Reply:
It's a tad bit more complicated than that. What we actually want is a type that has the [AsyncCommands](https://docs.rs/redis/0.23.0/redis/trait.AsyncCommands.html) and the `Clone` traits, as [ClusterClient](https://docs.rs/redis/0.23.0/redis/cluster/struct.ClusterClient.html) and [ConnectionManager](https://docs.rs/redis/0.23.0/redis/aio/struct.ConnectionManager.html) do.

I initially thought that a trait object was the answer, but then I found out that wouldn't work because `AsyncCommands` is only implemented for `Sized` types, and that breaks [object safety](https://doc.rust-lang.org/reference/items/traits.html#object-safety).

So, in the end, I used the same strategy we used for the [ConditionalSigner](https://github.com/cartesi/rollups/pull/124/files#diff-95a183c7c9178ec22d96a0c13f2e4c0e26f7fca31302d79de605c6604747d2c5R35) in #124.